WordPress/gutenberg
# Add link color and border theme_support

## What?

Updated February 8 2023.

The PR adds theme supports for link color and borders.

## Why?


1.  There is no way for themes to enabled these two features without theme.json (Theme.json can not be added to all classic themes without friction).
2. By using separate theme supports, theme developers can select which feature to support.
3. The default theme Twenty Twenty-One supported the previous link color theme support which was removed in 5.9. Following that, there have been bug reports on core trac about the link colors in the theme.

I have chosen to add all border settings as one theme support. This is different from how the settings in theme.json works.
In theme.json you enable the color, width, style and radius individually, please comment if you feel these should be separate theme supports.

Closes [41857](https://github.com/WordPress/gutenberg/issues/41857)
Alternative to [47451](https://github.com/WordPress/gutenberg/pull/47451)

Related: 
[core 57460 Backport: appearance-tools theme_support](https://core.trac.wordpress.org/ticket/57460)
[core 56487 Bundled themes: opt-in to appearance tools](https://core.trac.wordpress.org/ticket/56487)

## Testing Instructions
In your WordPress install, make sure that WP_DEBUG_DISPLAY, WP_DEBUG are set to true.
Activate Twenty Twenty-One.
You should see a doing it wrong message about `experimental-link-color`.
In your code editor, open functions.php in Twenty Twenty-One.
Around line 333, remove
`add_theme_support( 'experimental-link-color' );`

And add:
```
add_theme_support( 'link-color' );
add_theme_support( 'border' );
```

On refresh, the doing it wrong notice should no longer show.

Create a new post.
Add a group block with a paragraph inside. Add a link to the paragraph.
Confirm that the paragraph has a link color setting. Select a custom link color and confirm that it works.
_(Palette colors will not work on the front, that is a separate issue)_
Add a border to the group, test the different border style, width, and color. (select a custom color with the color picker)
and confirm that it works.

There are known bugs in Twenty Twenty-One that needs to be solved, you do not need to test them as part of this PR,
the main goal is to confirm that the controls are available.
